### PR TITLE
Removed references to etcdiscover tool

### PR DESCRIPTION
**Pull request by mhutchinson**
This was removed in trillian#2376. If this has value to anyone then it would be great to take a pull request to add this back. In the meantime, this at least removes dead references.

### CODE REVIEW

**Comment by roger2hk** [comment]
I find some more references to etcdiscover tool in this repo. See if you would like to remove them in this PR as well.

https://github.com/google/certificate-transparency-go/blob/ddf3c91f4732a18fe9855ab9cd649154e26354be/trillian/integration/prometheus.yml#L9

https://github.com/google/certificate-transparency-go/blob/86b3d313b44f7094b121e15e470689b3ec3cc803/.gitignore#L18

File: trillian/integration/ct_functions.sh [hunk 1]
```diff
@@ -63,17 +62,9 @@ ct_prep_test() {
...
         cat ${prom_cfg}
 
-        echo "Building etcdiscover"
-        go build github.com/google/trillian/monitoring/prometheus/etcdiscover
-
-        echo "Launching etcd service monitor updating ${etcdiscovered}"
-        ./etcdiscover ${ETCD_OPTS} --etcd_services=trillian-ctfe-metrics-http,trillian-logserver-http,trillian-logsigner-http -target=${etcdiscovered} --logtostderr &
-        ETCDISCOVER_PID=$!
         echo "Launching Prometheus (default location localhost:9090)"
         ${PROMETHEUS_DIR}/prometheus --config.file=${prom_cfg} \
```
Comment (roger2hk) [inline trillian/integration/ct_functions.sh:69]: If the content in `${prom_cfg}` is always empty, would there be an issue when the Prometheus is launched?

**Review by roger2hk — COMMENTED** [review]
I have got a question about the `${prom_cfg}`. LGTM for the rest.

File: trillian/integration/ct_functions.sh [hunk 2]
```diff
@@ -63,17 +62,9 @@ ct_prep_test() {
     if [[ ! -z "${ETCD_OPTS}" ]]; then
         PROMETHEUS_CFGDIR="$(mktemp -d ${TMPDIR}/ct-prometheus-XXXXXX)"
         local prom_cfg="${PROMETHEUS_CFGDIR}/config.yaml"
-        local etcdiscovered="${PROMETHEUS_CFGDIR}/trillian.json"
-        sed "s!@ETCDISCOVERED@!${etcdiscovered}!" ${CT_GO_PATH}/trillian/integration/prometheus.yml > "${prom_cfg}"
         echo "Prometheus configuration in ${prom_cfg}:"
         cat ${prom_cfg}
```
[the remark this answers is not in the document]
Comment (mhutchinson) [reply trillian/integration/ct_functions.sh:66]: Yup, I've removed this now.

File: trillian/integration/ct_functions.sh (hunk 1 above)
Comment (mhutchinson) [reply trillian/integration/ct_functions.sh]: Good spot. This is part of the problem from trying to fix up code that has bit-rotted. I've pointed this at the static config file now.

**Review by roger2hk — APPROVED** [review]

**Pull request merged**